src/main/java/pl/javastart/task/PriceCalc.java

```diff
@@ -0,0 +1,17 @@
+package pl.javastart.task;
+
+public class PriceCalc {
+    private final TaxRates taxRates = new TaxRates();
+
+    public double returnGrossPrice(Product product) {
+        if (product.getProductCategory() == null) {
+            return taxRates.grossPriceCalcTax3(product);
+        } else {
+            return switch (product.getProductCategory()) {
+                case Category.SWEETS -> taxRates.grossPriceCalcTax1(product);
+                case Category.DAIRY -> taxRates.grossPriceCalcTax2(product);
+                default -> taxRates.grossPriceCalcTax3(product);
+            };
+        }
+    }
```

Comment:
Tak jak pokazałem na filmie uprościłbym to rozwiązanie w ten sposób, żeby logika związana z obliczeniami znalazła się w tej metodzie i wtedy klasę TaxRates można usunąć